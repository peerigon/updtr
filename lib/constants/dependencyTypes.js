"use strict";

const REGULAR = 0;
const DEV = 1;
const OPTIONAL = 2;

function fromNpm(type) {
    switch (type) {
        case "dependencies":
            return REGULAR;
        case "devDependencies":
            return DEV;
        case "optionalDependencies":
            return OPTIONAL;
        default:
            throw new Error(`Unsupported dependency type: ${ type }`);
    }
}

exports.REGULAR = REGULAR;
exports.DEV = DEV;
exports.OPTIONAL = OPTIONAL;

exports.fromNpm = fromNpm;
// yarn uses the same strings. Change this if necessary.
exports.fromYarn = fromNpm;
