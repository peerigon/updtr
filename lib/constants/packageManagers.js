"use strict";

const NPM = "npm";
const YARN = "yarn";
const SUPPORTED = [NPM, YARN];

exports.NPM = NPM;
exports.YARN = YARN;
exports.SUPPORTED = SUPPORTED;
