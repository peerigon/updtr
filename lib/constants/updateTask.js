"use strict";

const WANTED = "wanted";
const LATEST = "latest";

exports.WANTED = WANTED;
exports.LATEST = LATEST;
