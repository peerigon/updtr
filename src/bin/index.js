import argv from "./argv";
import { create, run } from "./..";
import reporters from "../reporters";

async function start() {
    const cwd = process.cwd();
    const config = { ...argv };
    const reporterConfig = {
        stream: process.stdout,
        testStdout: argv.testStdout,
    };
    const reporter = reporters[argv.reporter];

    config.cwd = cwd;

    const updtr = create(config);

    reporter(updtr, reporterConfig);
    try {
        await run(updtr);
    } catch (err) {
        updtr.emit("error", err);
    }
}

start();
