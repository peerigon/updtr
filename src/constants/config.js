// The first value in arrays is the default value

export const USE_NPM = "npm";
export const USE_YARN = "yarn";
export const USE_OPTIONS = [USE_NPM, USE_YARN];
export const UPDATE_TO_LATEST = "latest";
export const UPDATE_TO_NON_BREAKING = "non-breaking";
export const UPDATE_TO_WANTED = "wanted";
export const UPDATE_TO_OPTIONS = [
    UPDATE_TO_LATEST,
    UPDATE_TO_NON_BREAKING,
    UPDATE_TO_WANTED,
];
export const SAVE_SMART = "smart";
export const SAVE_CARET = "caret";
export const SAVE_EXACT = "exact";
export const SAVE_OPTIONS = [SAVE_SMART, SAVE_CARET, SAVE_EXACT];
