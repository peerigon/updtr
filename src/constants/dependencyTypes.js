export const REGULAR = "dependencies";
export const DEV = "devDependencies";
export const OPTIONAL = "optionalDependencies";

export function fromNpm(type) {
    switch (type) {
        case "dependencies":
            return REGULAR;
        case "devDependencies":
            return DEV;
        case "optionalDependencies":
            return OPTIONAL;
        default:
            throw new Error(`Unsupported dependency type: ${ type }`);
    }
}

export {
    // yarn uses the same strings. Change this if necessary.
    fromNpm as fromYarn,
};
