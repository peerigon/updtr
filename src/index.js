import run from "./run";
import Updtr from "./Updtr";
import * as errors from "./errors";

export function create(config) {
    return new Updtr(config);
}

export { errors, run };
