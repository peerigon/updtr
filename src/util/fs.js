import fs from "fs";
import pify from "pify";

export default pify(fs);
