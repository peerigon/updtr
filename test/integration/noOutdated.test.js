import { USE_YARN } from "../../src/constants/config";
import {
    noOutdatedRegular as noOutdatedPackageContents,
} from "../fixtures/packageJsons";
import { create, run } from "../../src";
import getInstalledVersions, {
    filterUpdtrTestModule,
} from "./helpers/getInstalledVersions";
import {
    PACKAGE_JSON,
    YARN_LOCK,
    FIXTURE_NO_OUTDATED,
} from "./helpers/constants";
import { createTempDir, read, write } from "./helpers/fs";

// These tests may take longer on travis
jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;

describe("integration test: when there are no outdated dependencies", () => {
    describe("npm", () => {
        it("should not change the package.json, change the installed dependencies and resolve with an empty array as results", async () => {
            const tempDir = await createTempDir();

            await write(tempDir, PACKAGE_JSON, noOutdatedPackageContents);

            const updtr = create({
                cwd: tempDir,
            });
            const results = await run(updtr);
            const [
                tempDirPackageContents,
                installedVersions,
            ] = await Promise.all([
                read(tempDir, PACKAGE_JSON),
                getInstalledVersions(tempDir, filterUpdtrTestModule),
            ]);

            expect(results).toEqual([]);
            expect(tempDirPackageContents).toBe(noOutdatedPackageContents);
            expect(installedVersions.versions).toMatchSnapshot(
                "npm > installed versions"
            );
        });
    });
    describe("yarn", () => {
        it("should not change the package.json and yarn.lock and resolve with an empty array as results", async () => {
            const [tempDir, yarnLockContents] = await Promise.all([
                createTempDir(),
                read(FIXTURE_NO_OUTDATED, YARN_LOCK),
            ]);

            await write(tempDir, PACKAGE_JSON, noOutdatedPackageContents);

            const updtr = create({
                cwd: tempDir,
                use: USE_YARN,
            });
            const results = await run(updtr);
            const [
                tempDirPackageContents,
                installedVersions,
                tempDirYarnLockContents,
            ] = await Promise.all([
                read(tempDir, PACKAGE_JSON),
                getInstalledVersions(tempDir, filterUpdtrTestModule),
                read(tempDir, YARN_LOCK),
            ]);

            expect(results).toEqual([]);
            expect(tempDirPackageContents).toBe(noOutdatedPackageContents);
            expect(installedVersions.versions).toMatchSnapshot(
                "npm > installed versions"
            );
            expect(tempDirYarnLockContents).toBe(yarnLockContents);
        });
    });
});
