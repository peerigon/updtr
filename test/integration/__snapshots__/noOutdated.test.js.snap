// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`npm > installed versions 1`] = `
Object {
  "updtr-test-module-1": "2.0.0",
  "updtr-test-module-2": "2.1.1",
}
`;

exports[`npm > installed versions 2`] = `
Object {
  "updtr-test-module-1": "2.0.0",
  "updtr-test-module-2": "2.1.1",
}
`;
